mono/metadata/reflection.c

```diff
@@ -14539,13 +14540,13 @@ mono_reflection_call_is_assignable_to (MonoClass *klass, MonoClass *oklass)
 	g_assert (mono_class_get_ref_info (klass));
 	g_assert (!strcmp (((MonoObject*)(mono_class_get_ref_info (klass)))->vtable->klass->name, "TypeBuilder"));
 
-	params [0] = mono_type_get_object_checked (mono_domain_get (), &oklass->byval_arg, &error);
-	mono_error_raise_exception (&error); /* FIXME don't raise here */
+	params [0] = mono_type_get_object_checked (mono_domain_get (), &oklass->byval_arg, error);
+	return_val_if_nok (error, FALSE);
 
-	res = mono_runtime_try_invoke (method, (MonoObject*)(mono_class_get_ref_info (klass)), params, &exc, &error);
+	res = mono_runtime_try_invoke (method, (MonoObject*)(mono_class_get_ref_info (klass)), params, &exc, error);
 
-	if (exc || !mono_error_ok (&error)) {
-		mono_error_cleanup (&error);
+	if (exc || !mono_error_ok (error)) {
+		mono_error_cleanup (error);
 		return FALSE;
```

Comment:
Yes, if `IsAssignableTo` throws in managed and co-op is disabled, `error` will be ok and `exc` will be non-NULL.  If we fail to perform the invoke (either in coop or non-coop) for some reason, `exc` will be NULL and `error` will be not ok.